src/Bindings/ManualBindings_World.cpp

```diff
@@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
 		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
 	}
 
-	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
-	Plugin->AddResettable(task);
-	self->QueueTask(task);
+	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
+	Plugin->AddResettable(ResettableTask);
+	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
```

Comment:
`self->QueueTask([&ResettableTask] (cWorld & a_World) {ResettableTask->Run(a_World)});`
Or we can make cRessetable more supportive of the functional style, by adding:
```
template<class F>
void TryDoWithPlugin(F callback)
{
     cCSLock Lock(m_CSPlugin);
     if (m_Plugin != nullptr)
     {
          F(m_Plugin);
     }
}
```